Monsters hunting persistence/src/main/java/cz/fi/muni/pa165/dao/AreaDaoImpl.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package cz.fi.muni.pa165.dao;
+
+import cz.fi.muni.pa165.entity.Area;
+import cz.fi.muni.pa165.enums.AreaType;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.persistence.PersistenceContext;
+import static org.assertj.core.util.Preconditions.checkNotNull;
+
+/**
+ *
+ * @author Jan Gol <gol.honza@gmail.com>
+ */
+public class AreaDaoImpl implements AreaDao {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Override
+    public void create(Area area) {
+        checkNotNull(area);
+        entityManager.persist(area);
+    }
+
+    @Override
+    public void delete(Area area) {
+        checkNotNull(area);
+        entityManager.remove(area);
+    }
+
+    @Override
+    public Area findById(Long id) {
+        return entityManager.find(Area.class, id);
+    }
+
+    @Override
+    public Area findByName(String name) {
+        if (name == null) {
+            throw new IllegalArgumentException("Name cannot be null!");
+        }
+        try {
+            return entityManager.createQuery("SELECT a FROM Area a WHERE name = :name", Area.class)
+                    .setParameter("name", name)
+                    .getSingleResult();
+        } catch (NoResultException ex) {
+            return null;
+        }
+    }
+
+    @Override
+    public List<Area> getAllForType(AreaType areaType) {
+        return entityManager.createQuery("SELECT a FROM Area a WHERE type = :areaType", Area.class)
+                .getResultList();
```

Comment:
I believe that there should be `.addParameter("areaType", areaType)` called on the query before calling `getResultList()`.